packpack

```diff
@@ -18,6 +18,21 @@ DOCKER_REPO=${DOCKER_REPO:-packpack/packpack}
 # Extra parameters for docker run.
 PACKPACK_EXTRA_DOCKER_RUN_PARAMS="${PACKPACK_EXTRA_DOCKER_RUN_PARAMS:-}"
 
+# Look for podman vs docker
+# Note: if docker ever moves to rootless, this might be cleaned up
+set -e
+if type -p podman > /dev/null; then
+  # This might be needed for all SELinux installs
+  EXTRA_MOUNT=,Z
```

Comment:
Not sure it is safe default. At least source tree, cache and packpack directory directory may be intended to use from other containers and from the host system as well (it seems some services may lost access to those directories after labeling).

I read several related articles:

https://github.com/moby/moby/issues/30934
https://www.projectatomic.io/blog/2015/06/using-volumes-with-docker-can-cause-problems-with-selinux/

And now I'm not ever sure whether we'll able to run packpack on the same source tree several times. Or docker/podman will relabel it each time (remove old label and add a new one)? So only parallel builds will fail?

What do you think? Should not we make this choice explicit rather than default?

Reply:
I have no idea; I am not an SELinux expert. I just know that without it, I couldn't read the files.

Reply:
Just obserted the following call in the `packpack` script:

```sh
#
# Fix security context for selinux
#
chcon -Rt svirt_sandbox_file_t ${PACKDIR} ${SOURCEDIR} ${BUILDDIR} \
    1> /dev/null 2> /dev/null || :
```

AFAIU (I'm not an expert too), it doing the same (but like `,z`, not `,Z`) for all host directories that will be used as volumes except `${CACHE_DIR}`.

Can you share precise error? Maybe we should just add `${CACHE_DIR}` here?

Reply:
OK, when I comment out my `EXTRA_MOUNT`, this is the result:

```
+ docker run --volume /home/revragnarok/git_stuff/packpack/pack:/pack:ro --volume /home/revragnarok/git_stuff/packpack:/source:ro --volume /home/revragnarok/git_stuff/packpack/build:/build:rw --env-file /home/revragnarok/git_stuff/packpack/build/env --workdir /source --rm=true --tty=true --entrypoint=/build/userwrapper.sh -e XDG_CACHE_HOME=/cache -e CCACHE_DIR=/cache/ccache -e TMPDIR=/tmp -e CI= --volume /home/revragnarok/.cache/packpack:/cache:rw packpack/packpack:fedora-33 make -f /pack/Makefile -C /source BUILDDIR=/build -j
Emulate Docker CLI using podman. Create /etc/containers/nodocker to quiet msg.
/bin/sh: /build/userwrapper.sh: Permission denied

```

Reply:
OK, when I comment out my `EXTRA_MOUNT`, this is the result:

```
+ docker run --volume /home/user/git_stuff/packpack/pack:/pack:ro --volume /home/user/git_stuff/packpack:/source:ro --volume /home/user/git_stuff/packpack/build:/build:rw --env-file /home/user/git_stuff/packpack/build/env --workdir /source --rm=true --tty=true --entrypoint=/build/userwrapper.sh -e XDG_CACHE_HOME=/cache -e CCACHE_DIR=/cache/ccache -e TMPDIR=/tmp -e CI= --volume /home/user/.cache/packpack:/cache:rw packpack/packpack:fedora-33 make -f /pack/Makefile -C /source BUILDDIR=/build -j
Emulate Docker CLI using podman. Create /etc/containers/nodocker to quiet msg.
/bin/sh: /build/userwrapper.sh: Permission denied

```

Reply:
OK that SELinux line is already in there for me as well and doesn't seem to be doing anything.

Reply:
I tried only applying the "`,Z`" to the rw volume, and it still didn't work.

```
+ make -f /pack/Makefile -C /source BUILDDIR=/build -j
make: *** /source: Permission denied.  Stop.
```
Let me know if there's anything else you want me to test, otherwise, this is what needs to be done to work on Fedora hosts.

Reply:
That looks unexpected, because we set the label for the build directory. Is it on a new project clone or after previous run with `,Z`?